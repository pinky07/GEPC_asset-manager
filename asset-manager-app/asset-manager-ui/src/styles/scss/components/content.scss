.content {
  & > .row:first-child {
    border-bottom: 1px solid $green;
    margin-bottom: 20px;
  }

  .nav.nav-tabs {
    padding: 20px 0 0 0;
  }

  .nav-tabs {
    .nav-link {
      border: none;
      color: white;
      background-color: $blue-1;
      font-weight: bold;
      padding: 4px 12px 12px 12px;

      &.active,
      &:hover,
      &:focus {
        border: none;
        color: white;
        background-color: $green;
      }
    }
  }
}
