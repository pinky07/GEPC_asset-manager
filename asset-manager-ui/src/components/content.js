import {
  Col,
  Container,
  Nav,
  NavItem,
  NavLink,
  Row,
  TabContent,
  TabPane,
} from 'reactstrap';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import React from 'react';

import GridView from './gridView/gridView';
import Menu from './menu';
import TreeView from './treeView/treeView';
import {
  getAssetsAllocation,
  getAllocationTree,
  showGridView,
  showTreeView,
} from '../actions';

const TREE_VIEW_TAB = '1';
const GRID_VIEW_TAB = '2';

/**
 * Content component with Grid and Tree View tabs.
 * 
 * @author Francisco ZuÃ±iga
 * @class Content
 * @extends {React.Component}
 */
export class Content extends React.Component {
  /**
   * Creates an instance of Content.
   * 
   * @param {any} props 
   * @memberof Content
   */
  constructor(props) {
    super(props);
    this.state = {
      activeTab: TREE_VIEW_TAB,
    };
  }

  /**
   * Toggles between Tree and Grid View
   * 
   * @memberof Content
   */
  toggle = index => {
    this.setState({
      activeTab: index,
    });
  };

  /**
   * Calls the action to get assets allocation
   */
  componentDidMount() {
    this.props.getAssetsAllocation();
  }

  /**
   * Checks for new assets allocation being passed as props
   * Sets the active view according to the assets allocation data
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.assetsAllocation !== this.props.assetsAllocation) {
      this.props.getAllocationTree();
    }
    if (nextProps.isTreeViewActive !== this.props.isTreeViewActive) {
      const activeTab = nextProps.isTreeViewActive
        ? TREE_VIEW_TAB
        : GRID_VIEW_TAB;
      this.toggle(activeTab);
    }
  }

  /**
   * Renders the component
   */
  render() {
    const { isLoading } = this.props;

    const treeViewSelected = this.state.activeTab === TREE_VIEW_TAB;
    const treeViewTabClass =
      this.state.activeTab === TREE_VIEW_TAB ? 'active' : '';
    const gridViewTabClass =
      this.state.activeTab === GRID_VIEW_TAB ? 'active' : '';

    const toggleTreeView = () => this.props.showTreeView();
    const toggleGridView = () => this.props.showGridView();
    return (
      <Container className="content" fluid>
        {isLoading ? <div className="loader" /> : null}
        <Row>
          <Col xs="3">
            <Nav tabs>
              <NavItem>
                <NavLink className={treeViewTabClass} onClick={toggleTreeView}>
                  Tree View
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink className={gridViewTabClass} onClick={toggleGridView}>
                  Grid View
                </NavLink>
              </NavItem>
            </Nav>
          </Col>
          <Col xs="9">
            <Menu treeViewSelected={treeViewSelected} />
          </Col>
        </Row>
        <Row>
          <Col xs="12">
            <TabContent activeTab={this.state.activeTab}>
              <TabPane tabId={TREE_VIEW_TAB}>
                <TreeView />
              </TabPane>
              <TabPane tabId={GRID_VIEW_TAB}>
                <GridView />
              </TabPane>
            </TabContent>
          </Col>
        </Row>
      </Container>
    );
  }
}

Content.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  isTreeViewActive: PropTypes.bool.isRequired,
  showTreeView: PropTypes.func.isRequired,
  showGridView: PropTypes.func.isRequired,
  assetsAllocation: PropTypes.object.isRequired,
  getAssetsAllocation: PropTypes.func.isRequired,
  getAllocationTree: PropTypes.func.isRequired,
};

Content.defaultProps = {
  isLoading: false,
};

const mapStateToProps = state => {
  return {
    ...state.shared,
  };
};

export default connect(mapStateToProps, {
  getAssetsAllocation,
  getAllocationTree,
  showGridView,
  showTreeView,
})(Content);
