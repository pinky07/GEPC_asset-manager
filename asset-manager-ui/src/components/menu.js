import { Button } from 'reactstrap';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import React from 'react';

import {
  copyTree,
  copyTreeWithoutManager,
  deleteTree,
  importMix,
  loadIFMarketValues,
  loadIFTree,
  newTree,
  preview,
} from '../actions';

/**
 * Displays a menu
 * @export
 * @class Menu
 * @extends {React.Component}
 */
export class Menu extends React.Component {
  /**
   * Renders this component
   * @returns Rendered HTML
   * @memberof Menu
   */
  render() {
    const { treeViewSelected } = this.props;
    const { data } = this.props.tree;
    const treeWithData = data && data.length > 0;

    let buttons = [];
    if (treeViewSelected) {
      buttons = [
        {
          name: 'Load IF Tree',
          function: this.props.loadIFTree,
          disabled: treeWithData,
        },
        {
          name: 'Load IF Market Values',
          function: this.props.loadIFMarketValues,
          disabled: false,
        },
        {
          name: 'Copy Tree',
          function: this.props.copyTree,
          disabled: false,
        },
        {
          name: 'Copy Tree (w/o managers)',
          function: this.props.copyTreeWithoutManager,
          disabled: false,
        },
        {
          name: 'Delete Tree',
          function: this.props.deleteTree,
          disabled: !treeWithData,
        },
        {
          name: 'New Tree',
          function: this.props.newTree,
          disabled: treeWithData,
        },
      ];
    } else {
      buttons = [
        { name: 'Preview', function: this.props.preview, disabled: false },
        { name: 'Import Mix', function: this.props.importMix, disabled: false },
      ];
    }

    return (
      <div className="app-menu">
        <div className="buttons">
          {buttons.map((button, index) => (
            <Button
              key={index}
              color="primary"
              onClick={button.function}
              disabled={button.disabled}>
              {button.name}
            </Button>
          ))}
        </div>
      </div>
    );
  }
}

Menu.propTypes = {
  treeViewSelected: PropTypes.bool.isRequired,
  tree: PropTypes.object.isRequired,
  copyTree: PropTypes.func.isRequired,
  copyTreeWithoutManager: PropTypes.func.isRequired,
  deleteTree: PropTypes.func.isRequired,
  importMix: PropTypes.func.isRequired,
  loadIFMarketValues: PropTypes.func.isRequired,
  loadIFTree: PropTypes.func.isRequired,
  newTree: PropTypes.func.isRequired,
  preview: PropTypes.func.isRequired,
};

/**
 * Maps relevant properties to this component from the store.
 * 
 * @param {*} state 
 */
const mapStateToProps = state => {
  return {
    ...state.allocationTree,
  };
};

/**
 * Connects this component to Redux
 */
export default connect(mapStateToProps, {
  loadIFTree,
  loadIFMarketValues,
  copyTree,
  copyTreeWithoutManager,
  deleteTree,
  newTree,
  preview,
  importMix,
})(Menu);
