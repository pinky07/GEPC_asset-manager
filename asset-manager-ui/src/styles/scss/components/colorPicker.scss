/**
 * colorPicker.scss
 * 
 * Handles the style of the Color Picker inside the Node Details Section
 *
 *
 * @author Francisco Zuñiga
 * @author Rubén Jiménez
 */

.colorPickerWrapper {
  width: 100%;
  height: 2.25rem; // Has to match with the colorPicker inside

  .colorPicker {

    // This matches the Bootstrap style of an Input next to an InputGroupAddon
    height: 2.25rem; 
    border-top-left-radius: 0;
    border-top-right-radius: 0.25rem;
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0.25rem;
    
    // Style
    margin: 0;
    padding: 0;
    background: white;
    display: inline-block;
    cursor: pointer;
    width: 20%;
    border-left: none;
    
    .color {
      // This width and height consider the 1 pixel border
      height: 2.25rem;
      width: 2.25rem;
      width: 100%;
      border-top-left-radius: 0;
      border-top-right-radius: 0.25rem;
      border-bottom-left-radius: 0;
      border-bottom-right-radius: 0.25rem;
      margin-left: auto;
    }

    .popover {
      position: absolute;
      left: 11em;
      top: -9em;
      z-index: 99;
    }

    .cover {
      position: fixed;
      bottom: 0px;
      top: 0px;
      left: 0;
      right: 0;
    }

    &.disabled {
      background-color: #e9ecef;
      opacity: 1
    }
  }

  &.disabled .colorPicker { 
    background-color: #e9ecef;
    opacity: 1;
    cursor: not-allowed;
  }
}