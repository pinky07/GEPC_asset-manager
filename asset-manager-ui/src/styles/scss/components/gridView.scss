.gridView {
  .dropdownPlanAnalysis {
    margin-left: 350px;
    width: 184px;
    display: inline-block;
  }

  .gridContainer {
    padding-top: .5em;
    width: 100%;
    height: 800px;
    padding-bottom: 2em;
    overflow: hidden;
  }

  .ag-header-row {
    color: white;
    font-weight: bold;
    background-color: $blue-1 !important;
  }

  .ag-header-cell-label {
    align-items: center;
    justify-content: center;
    height: 100%;
  }

  .ag-row.ag-row-even {
    background-color: $green-1;
  }

  .ag-row.ag-row-odd {
    background-color: $green-2;
  }

  .ag-react-container {
    .nav-link {
      padding: 0;
    }
  }

  i {
    display: inline-block;
    color: white;
    margin-left: 0.5em;
  }
}
