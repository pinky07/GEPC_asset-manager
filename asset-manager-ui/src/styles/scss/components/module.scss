/**
 * module.scss - version 1.0.0
 * 
 * Imports every other scss file in this folder
 *
 * @author Francisco Zu√±iga
 */

@import 'gridView';
@import 'treeView';
@import 'colorPicker';
@import 'content';
@import 'expanderFormatter';
@import 'header';
@import 'menu';
@import 'nodeDetails';
@import 'nodeMenu';
@import 'treeNodeRenderer';
