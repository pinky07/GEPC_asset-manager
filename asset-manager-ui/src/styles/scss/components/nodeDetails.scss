.nodeDetails {
  & > .card {
    padding: 20px;
  }

  .input-group {
    margin-bottom: 15px;
  }

  .input-group-addon {
    background-color: $blue-1;
    color: white;
    border: none !important;
    font-weight: bold;
  }

  .card-title {
    text-align: center;
    margin-bottom: 1em;
  }

  input:disabled,
  select:disabled {
    cursor: not-allowed
  }
}
