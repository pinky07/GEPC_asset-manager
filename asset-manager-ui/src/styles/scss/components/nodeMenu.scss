.nodeMenu {
  input {
    display: inline-block;
    width: 65px;
    height: 36px;
    margin-left: 1em;
    border-color: $green;
  }
}

.react-contexify {
  z-index: 100;
}

.react-contexify__item__data {
  width: 100%;
}