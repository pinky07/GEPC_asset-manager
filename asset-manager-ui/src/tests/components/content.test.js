import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { Content } from '../../components/content';
import allocation from '../../data/allocations.json';

describe('content component', () => {
  const props = {
    isLoading: false,
    isTreeViewActive: false,
    showTreeView: jest.fn(),
    showGridView: jest.fn(),
    assetsAllocation: allocation,
    getAssetsAllocation: jest.fn(),
    getAllocationTree: jest.fn(),
    getAllocationGrid: jest.fn(),
  };
  let wrapper = shallow(<Content {...props} />);

  it('renders correctly', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
