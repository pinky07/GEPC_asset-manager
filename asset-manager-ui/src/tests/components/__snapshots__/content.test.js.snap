// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content component renders correctly 1`] = `
<Container
  className="content"
  fluid={true}
  tag="div"
>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="3"
    >
      <Nav
        tabs={true}
        tag="ul"
      >
        <NavItem
          tag="li"
        >
          <NavLink
            className="active"
            onClick={[Function]}
            tag="a"
          >
            Tree View
          </NavLink>
        </NavItem>
        <NavItem
          tag="li"
        >
          <NavLink
            className=""
            onClick={[Function]}
            tag="a"
          >
            Grid View
          </NavLink>
        </NavItem>
      </Nav>
    </Col>
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="9"
    >
      <Connect(Menu)
        treeViewSelected={true}
      />
    </Col>
  </Row>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="12"
    >
      <TabContent
        activeTab="1"
        tag="div"
      >
        <TabPane
          tabId="1"
          tag="div"
        >
          <Connect(TreeView) />
        </TabPane>
        <TabPane
          tabId="2"
          tag="div"
        >
          <Connect(GridView) />
        </TabPane>
      </TabContent>
    </Col>
  </Row>
</Container>
`;
